narwhals/_spark_like/utils.py

```diff
@@ -58,6 +62,45 @@ def native_to_narwhals_dtype(
     return dtypes.Unknown()
 
 
+def narwhals_to_native_dtype(
+    dtype: DType | type[DType], version: Version
+) -> pyspark_types.DataType:
+    from pyspark.sql import types as pyspark_types
+
+    dtypes = import_dtypes_module(version)
+
+    if isinstance_or_issubclass(dtype, dtypes.Float64):
+        return pyspark_types.DoubleType()
+    if isinstance_or_issubclass(dtype, dtypes.Float32):
+        return pyspark_types.FloatType()
+    if isinstance_or_issubclass(dtype, dtypes.Int64):
+        return pyspark_types.LongType()
+    if isinstance_or_issubclass(dtype, dtypes.Int32):
+        return pyspark_types.IntegerType()
+    if isinstance_or_issubclass(dtype, dtypes.Int16):
+        return pyspark_types.ShortType()
+    if isinstance_or_issubclass(dtype, dtypes.Int8):
+        return pyspark_types.ByteType()
+    if isinstance_or_issubclass(dtype, dtypes.String):
+        return pyspark_types.StringType()
+    if isinstance_or_issubclass(dtype, dtypes.Boolean):
+        return pyspark_types.BooleanType()
+    if isinstance_or_issubclass(dtype, dtypes.Datetime):
+        return pyspark_types.TimestampType()
```

Comment:
a bit unsure if we should add this given that `test_cast_datetime_tz_aware` is not working yet. 

Should we raise a `NotImplementedError`?

Reply:
I would keep it out for now, as this implementation might lead to the wrong outcome if Datetime with timezone is provided.